data/virt_autotest/virt_logs_collector.sh

```diff
@@ -414,21 +414,54 @@ guest_hash_index=0
 guest_current=""
 dhcpd_lease_file="/var/lib/dhcp/db/dhcpd.leases"
 
+#Install necessary packages
+echo -e "Install necessary packages. zypper install -y sshpass nmap xmlstarlet" | tee -a ${virt_logs_collecor_log}
+zypper install -y sshpass nmap xmlstarlet | tee -a ${virt_logs_collecor_log}
+
+#Establish reachable networks and hosts database on host
+subnets_in_route=`ip route show all | awk '{print $1}' | grep -v default`
+subnets_scan_results=""
+subnets_scan_index=0
+echo -e "Subnets ${subnets_in_route[@]} are reachable on host judging by ip route show all" | tee -a ${virt_logs_collecor_log}
+echo -e "Establishing reachable hosts in subnets ${subnets_in_route[@]} database on host" | tee -a ${virt_logs_collecor_log}
+for single_subnet in ${subnets_in_route[@]};do
+    single_subnet_transformed=${single_subnet//./_}
+    single_subnet_transformed=${single_subnet_transformed/\//_}
+    scan_timestamp=`date "+%F-%H-%M-%S"`
+    mkdir -p "${virt_logs_folder}/nmap_subnets_scan_results"
+    single_subnet_scan_results=${virt_logs_folder}'/nmap_subnets_scan_results/nmap_scan_'${single_subnet_transformed}'_'${scan_timestamp}
+    subnets_scan_results[${subnets_scan_index}]=${single_subnet_scan_results}
+    echo -e "nmap -sn $single_subnet -oX $single_subnet_scan_results" | tee -a ${virt_logs_collecor_log}
+    nmap -sn $single_subnet -oX $single_subnet_scan_results  | tee -a ${virt_logs_collecor_log}
+    subnets_scan_index=$(( ${subnets_scan_index} + 1 ))
+done
+
```

Comment:
Seems similar code block for "install package/establish reachable network and host database at host" repeating in this file and fetch_logs_from_guest.sh. Is it possible to figure out someway to share code?

Reply:
This file and fetch_logs_from_guest.sh are meant to be separate and standalone scripts. So they can run totally independently. I do think there is a need to make them share something. Additionally, I just submitted the change together, so you see "this kind of duplication". I do not want to introduce extra something mechanism to further complicate things here.